### DISCUSSION TITLE
Need Help in getting number of all Pull Requests (across all repos) for an organization per month

### DISCUSSION THREAD

**Discussion by zoaib** [community/community#24564] [Apps, API and Webhooks]
<p>Hello All, I am new to github API’s so looking for your help/guidance. I want to get the list of all pull requests across all private (organization) repos within a specific time duration (monthly). Can anyone proivde guidance how I can get this data through API’s.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/zoaib">@zoaib</a>,</p>
<p>Thank you for being here! You can use our API to query that information, for example:</p>
<pre><code>{
  search(query: "org:atom is:pr is:open created:&gt;2020-04-01", type: ISSUE, last: 100) {
    edges {
      node {
        ... on PullRequest {
          url
          title
          createdAt
        }
      }
    }
  }
}
</code></pre>
<p>The search query above, would return the last 100 pull requests in the <code>atom</code> organization created after April first. You can give it a go in the <a href="https://developer.github.com/v4/explorer/" rel="nofollow noopener">explorer</a>. </p>
<p>I’d also recommend checking out our <a href="https://developer.github.com/v4/" rel="nofollow noopener">API documentation</a> to get familiarized with available endpoints, schemas etc.</p>

**Reply by tcolonel** [reply]
> The search query above, would return the last 100 pull requests in the atom organization created after April first.

What about the same but for project and 2 dates - start and end?

Thanks.

**Reply by AndreaGriffiths11** [reply]
You must provide a `first` or `last` value to properly paginate the `search` connection, but first, you run your GraphQL query without a cursor. This query will return the first set of results and pagination information. For example, your initial query might look like this:

```
{
  search(query: "org:atom is:pr created:2018-11-11..2019-11-11", type: ISSUE, first: 100) {
    edges {
      node {
        ... on PullRequest {
          url
          title
          createdAt
        }
      }
    }
    pageInfo {
      endCursor
      hasNextPage
    }
  }
}
```

The response to this query will include both the data you requested (edges) and pagination information (pageInfo). The pageInfo object contains the endCursor and a boolean hasNextPage.

Extract the Cursor: Look for the endCursor value in the pageInfo part of the response. This value is the cursor you will use for your next query. It will look like a string of characters, for example: "Y3Vyc29yOnYyOpHOABXkFw==".

Use the Cursor in Subsequent Queries: For your next query, you'll include this cursor in the after parameter to get the next set of results. Here’s how you'd modify the query:

```
{
  search(query: "org:atom is:pr created:2018-11-11..2019-11-11", type: ISSUE, first: 100, after: "Y3Vyc29yOnYyOpHOABXkFw==") {
    edges {
      node {
        ... on PullRequest {
          url
          title
          createdAt
        }
      }
    }
    pageInfo {
      endCursor
      hasNextPage
    }
  }
}
```

Replace "Y3Vyc29yOnYyOpHOABXkFw==" with the actual cursor from your previous response.
You can continue this process, updating the cursor each time, until hasNextPage is false, indicating there are no more pages to fetch. This allows you to paginate through large sets of data in a manageable way. Remember, each cursor is specific to the query and the state of the data at the time of the query.
